rotkehlchen/chain/ethereum/decoding/kyber/decoder.py

```diff
@@ -0,0 +1,73 @@
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple
+
+from rotkehlchen.accounting.structures import (
+    HistoryBaseEntry,
+    HistoryEventSubType,
+    HistoryEventType,
+)
+from rotkehlchen.chain.ethereum.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.ethereum.decoding.structures import ActionItem
+from rotkehlchen.chain.ethereum.structures import EthereumTxReceiptLog
+from rotkehlchen.chain.ethereum.types import string_to_ethereum_address
+from rotkehlchen.chain.ethereum.utils import ethaddress_to_asset
+from rotkehlchen.types import ChecksumEthAddress, EthereumTransaction
+from rotkehlchen.user_messages import MessagesAggregator
+from rotkehlchen.utils.misc import hex_or_bytes_to_address
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.ethereum.manager import EthereumManager
+
+TOKEN_SWAP = b"\x18I\xbdj\x03\n\x1b\xca(\xb847\xfd=\xe9o='\xa5\xd1r\xfa~\x9cx\xe7\xb6\x14h\x92\x8a9"  # noqa: E501
+
+
+class KyberDecoder(DecoderInterface):
+    def __init__(  # pylint: disable=super-init-not-called
+            self,
+            ethereum_manager: 'EthereumManager',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: MessagesAggregator,
+    ) -> None:
+        self.ethereum_manager = ethereum_manager
+        self.base = base_tools
+        self.msg_aggregator = msg_aggregator
+
+    def _decode_old_swap(  # pylint: disable=no-self-use
+            self,
+            tx_log: EthereumTxReceiptLog,
+            transaction: EthereumTransaction,  # pylint: disable=unused-argument
+            decoded_events: List[HistoryBaseEntry],  # pylint: disable=unused-argument
+            all_logs: List[EthereumTxReceiptLog],  # pylint: disable=unused-argument
+    ) -> Tuple[Optional[HistoryBaseEntry], Optional[ActionItem]]:
+        source_token_address = hex_or_bytes_to_address(tx_log.data[:32])
+        source_token = ethaddress_to_asset(source_token_address)
```

Comment:
no this is not enough. You should find the exact same event. Which means token, amount, source and generally verify as many ofthe fiellds as possible.

Also where is the destination/to token?

Check out the uniswap or 1inch-v1 and 1inch-v2 swap decoding to understand.

Your example here would fail here and pick any random ERC transfer for the "receive" part and also for the spend part if amount, or address is not the same